Add tests to Tdarr_Plugin_TN10_tehNiemer_extract_remove_subs

tehniemer opened the pull request.

tehniemer committed:
Remove tdarrSkipTest

tehniemer committed:
Add tests file

tehniemer committed:
lint fixes

tehniemer committed:
lint fixes

tehniemer committed:
fix issue in test

tehniemer committed:
Update Tdarr_Plugin_TN10_SUBS.js

reorder checks

tehniemer committed:
Update Tdarr_Plugin_TN10_SUBS.js

test fixes

tehniemer committed:
Update Tdarr_Plugin_TN10_SUBS.js

lint fixes

tehniemer committed:
Update Tdarr_Plugin_TN10_SUBS.js

added additional subtitle dispositions

tehniemer committed:
Update Tdarr_Plugin_TN10_SUBS.js

modify method of checking for undefined dispositions

tehniemer commented:
@haveagitgat would you mind taking a look at this failure and let me know what you think is going on? It's not clear to me what exactly is failing with the test or how I can go about debugging this.  

I'm checking that each variable is defined in the script before the value is checked.

```
Error: Command failed: node "/home/runner/work/Tdarr_Plugins/Tdarr_Plugins/tests/Community/Tdarr_Plugin_TN10_SUBS.js"
Error: Unexpected plugin error!TypeError: Cannot read properties of undefined (reading 'file')
    at run (/home/runner/work/Tdarr_Plugins/Tdarr_Plugins/tests/helpers/run.js:58:17)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

[linux] Tdarr_Plugin_TN10_SUBS.js: test 0
[linux] Tdarr_Plugin_TN10_SUBS.js: test 1
[linux] Tdarr_Plugin_TN10_SUBS.js: test 2

Error: Unexpected plugin error!TypeError: Cannot read properties of undefined (reading 'file')
    at run (/home/runner/work/Tdarr_Plugins/Tdarr_Plugins/tests/helpers/run.js:58:17)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

Error: Process completed with exit code 1.
```

HaveAGitGat commented:
`[linux] Tdarr_Plugin_TN10_SUBS.js: test 2`

So error occured in 3rd test (starting from index 0)

Replace the /tests/helpers/run.js code with this code and it will print the stack error with the exact line the error occurred:
https://pastebin.com/tbM72zhy

I'll add that change to the main repo.

So it says error occured on line 275 of your plugin which is to do with the originalLibraryFile which doesn't exist in the plugin test inputs (not needed for most plugins).

So update the third test input to:

```
    input: {
      file: _.cloneDeep(require('../sampleData/media/sampleH264_2.json')),
      librarySettings: {},
      inputs: {
        language: 'eng',
      },
      otherArguments: {
        originalLibraryFile:_.cloneDeep(require('../sampleData/media/sampleH264_2.json')),
      },
    },
```

HaveAGitGat commented:
Added the change to master here: https://github.com/HaveAGitGat/Tdarr_Plugins/pull/573

tehniemer committed:
Update Tdarr_Plugin_TN10_SUBS.js

test fixes

tehniemer committed:
Update Tdarr_Plugin_TN10_SUBS.js

lint fixes

tehniemer committed:
Update Tdarr_Plugin_TN10_SUBS.js

chaged `originalLibraryFile` to `file` in plugin and added test

tehniemer committed:
lint fixes

tehniemer committed:
Update Tdarr_Plugin_TN10_SUBS.js

eslint-disable-next-line @typescript-eslint/

tehniemer committed:
Update Tdarr_Plugin_TN10_SUBS.js

lint fixes

tehniemer committed:
test fixes and updates

tehniemer committed:
test fixes

tehniemer committed:
Update Tdarr_Plugin_TN10_SUBS.js

output fixes

tehniemer commented:
Thank you so much! your guidance has helped massively.

tehniemer committed:
Update and rename Tdarr_Plugin_TN10_SUBS.js to Tdarr_Plugin_TN10_tehNiemer_extract_remove_subs.js

tehniemer committed:
Rename Tdarr_Plugin_TN10_SUBS.js to Tdarr_Plugin_TN10_tehNiemer_extract_remove_subs.js

tehniemer committed:
Update Tdarr_Plugin_TN10_tehNiemer_extract_remove_subs.js

tehniemer committed:
Bring up to date with head (#29)

* Restored some cases for HW decode but commented out

* Quick update to tooltip

* Protection against errors if unable to use MediaInfo or ffprobe

Just changed the videoBR logic to be inside a try block to avoid errors on rare instances where Media info & ffprobe fail

* Created Flow plugin for Radarr and Sonarr renaming

* Compiled Flow plugin for Radarr and Sonarr renaming

* Rename with radarr/sonarr Flow plugin now updates the file id

* Force rename with radarr/sonarr : corrected mistake

* Force rename with radarr/sonarr : handling no rename needed

* Force rename with radarr/sonarr : improve readability

* Force rename with radarr/sonarr : corrected mistake

* Force rename with radarr/sonarr : added file move after rename

* Force rename radarr.sonarr : remove log

* Force rename Radarr/Sonarr logic simplified to just apply the naming policy

* Added some logs for compare file size

* Enhanced logs for campare file size

* Apply radarr/sonarr naming policy : handled file not found use case and enforced some strong typing

* Found new edge cases so changed how HW decode is triggered

Ended up finding more cases where errors would pop up and looks to all to do with HW decode.
- Now ensured that unless the incoming format is in the white list then sw decode is done instead
- Correctly removed -hwaccel cmd when sw decode happens preventing accidental hwdecode
- Specify the format to decode when using sw
- renamed oldformat var to match its use case (now swDecode)
- Info log updated with when SW decode happens and will report which incoming format caused it

* Revert "Added some logs for compare file size"

This reverts commit b62e38bfde13f83708149b541787a370a90d4b96.

* Variables and functions names refactoring

* Little more refactoring

* Noticed one last -vf filter check used a colon incorrectly

Changed to a comma which matches all other formatting with the -vf cmd & updated tests

* Final things, remove tag mkvpropedit & bump the Version number ;)

* Renamed some methods to clarify the expected behavior

* Added new output to the plugin for the case when file is not found

* Updates to video loop & darwin support

- Corrected some Mac/Darwin support, incorrect 10 bit options (In future will remove videotoolbox support and split into separate plugin)
- Improved video check loops, some logic did seem to not correctly stop when needed & the remove stream option had the wrong syntax so probably wouldn't have worked

* Corrected final check

* Final test completed

* Added more checks and fallbacks for name checking

* Minor correction

* Removed useless log

* Refactoring + more robust + added safeties and a new output for file not found

* Added HDR support, Cleaned up info log & initial test fix

- Added HDR support. Tested as working but can have issues. Warning applied
- Made duration check a bit better with try statement
- Cleaned up how the info log was used so it's more consistent & fixed up trailing spaces & line breaks
- Initial test fix up but more will be needed

* Added new version for checkFileNameIncludes flow plugin checking for regex patterns

* Escaping regex special caracters

* Removed the possibility for regex patterns to be a list. All patterns must be merged

* Fix eslint warnings and errors

* More eslint fixes

* Fix eslint warnings and errors

* Fix eslint warnings and errors

* Replaced custom code by array.prototype.find

* Replaced any by inferred interface

* Replaced any by inferred interfaces

* Commas added where missing

* Added imdbid detection from file name to get the movie or serie id

* Corrected lookup endpoint param name

* Log corrected

* Addded lookup url for Sonarr

* Serie lookup url corrected

* Multiple small adjustments

* Corrected log position

* Final corrections after testing

* Removed useless dependency

* Added the possibility to fin movieId from imdbId

* Log corrected

* Minor corrections

* Added the possibility to find serieId from imdbId

* After thorough testing, some renaming and refactoring

* After thorough testing, some renaming and refactoring

* Icon changed to fa-pen-to-square

* Made imdb regex ungreedy

* Made imdb regex ungreedy

* Make case insensitive

* Update Tdarr_Plugin_bsh1_Boosh_FFMPEG_QSV_HEVC.js test

* Update Tdarr_Plugin_bsh1_Boosh_FFMPEG_QSV_HEVC.js test

* Add handBrakeMode: false,

* Further test fixes for linux env

* Update test logic to display all test errors

* Update Tdarr_Plugin_bsh1_Boosh_FFMPEG_QSV_HEVC.js darwin tests

* Apply auto-build changes

* Log tsc version

* Option to use perc bitrate

* Apply auto-build changes

* Check against multiple values for ==

* Apply auto-build changes

* Add multi value to !=

* Apply auto-build changes

* Update checkout actions

* Remove unused eslint rule

* Simplify logic and readability

* github.head_ref

* Don't auto build on fork PR

* Don't autobuild on fork PR

* Use github.event.pull_request.head.sha

* Add compiled changes

* Fix Sonarr typo

* Fix Sonarr typo

* Add compiled changes

* Add compiled changes

* Use templating

* Add brackets

* Add compiled changes

* CPU and GPU Workers with Classic Plugins

* Update fork check

* Apply auto-build changes

* Set Flow Variable

* Ignore lint errors

* Fix Check Video Resolution typo

* Apply auto-build changes

* Adjust desc

* Apply auto-build changes

* Pause/Unpause All Nodes

* Apply auto-build changes

* Set original after adding typeIndex

* Apply auto-build changes

* Add Run CLI

* Apply auto-build changes

* Update logic

* Apply auto-build changes

* Update var to userCli

* Apply auto-build changes

* Add Apprise

* Format tooltip

* Apply auto-build changes

* Auto build

* Apply auto-build changes

* Add name

* Fix yml syntax error

* Format description

* Apply auto-build changes

* Add  'Auto accept successful transcodes' note

* Apply auto-build changes

* Add 2nd classic transcode plugin output for if no processing occurred

* Apply auto-build changes

* Remove default

* Update tooltip

* Apply auto-build changes

* Check mediaInfo properties

* Update description

* Apply auto-build changes

* Add additional compare logging

* Format text

* Apply auto-build changes

* Add useCustomCliPath and customCliPath inputs

* Use destructuring

* Apply auto-build changes

* Format classic plugin names

* Use Node scan function if available

* Apply auto-build changes

* Add types

* Add version comment

* Flow Run Health Check updates stats and queues

* Apply auto-build changes

* Fix error message with cliPath

* Apply auto-build changes

* Add files via upload

* Add files via upload

* Update worker with hb fps if available

* Apply auto-build changes

* Use fsp for promises

* Ensure es6 import

* Use async fileExists

* Promisify fs.unlinkSync

* Promisify copyFileSync

* Apply auto-build changes

* Fix rename-undo bug when using rename plugin

* Apply auto-build changes

* Use simpler map trim() for values

* Fix comment typo

* Fix genpts comment typo

* Add tags plugins

* Update inferfaces

* Apply auto-build changes

* Don't use scale_qsv=format=p010le on Windows

* Apply auto-build changes

* Tags subset fix

* Apply auto-build changes

* Update tag plugin descriptions

* Apply auto-build changes

* Trim and filter tags

* Apply auto-build changes

* Compare File Size Ratio Live

* Apply auto-build changes

* Update tooltip

* Apply auto-build changes

* Add Basic HEVC Video Flow - Migz 50%

* Apply auto-build changes

* Add error variable

* Apply auto-build changes

* Add original container option.

* Add toggles for preset and quality

* Apply auto-build changes

* Adjust input labels

* Apply auto-build changes

* Log CLI exit code

* Apply auto-build changes

* Add ffmpegCommand.init to ensure "Begin" command has been used

* Add checkFfmpegCommandInit(() to ffmpeg command plugins

* Apply auto-build changes

* Use checkFfmpegCommandInit() in Execute plugin

* Apply auto-build changes

* Re-order filepath change logging

* Use macos-14

* Use windows-2022

---------

Co-authored-by: Boosh1 <45874141+Boosh1@users.noreply.github.com>
Co-authored-by: jeanchristophe.mqt@gmail.com <jeanchristophe.mqt@gmail.com>
Co-authored-by: HaveAGitGat <43864057+HaveAGitGat@users.noreply.github.com>
Co-authored-by: HaveAGitGat <HaveAGitGat@users.noreply.github.com>
Co-authored-by: tws101 <tws101@users.noreply.github.com>
